### ISSUE TITLE
[Feature Request] Split and truncate trajectories

### ISSUE THREAD

**Issue by smorad** [pytorch/rl#1428]
## Motivation

For sequence models/recurrent RL, we often want a long sequence to be split and padded into equal-sized segments of shape `(batch, segment_length)`. `split_trajectories` handles the padding, but does not account for splitting long sequences (i.e. sequence length > segment length should be split into multiple segments). I propose that I add such a method to tensordict.

## Solution
First, we need to add `torch.repeat_interleave` support to tensordict. Then, we can do something like
```python
def truncate_trajectories(td, segment_length):
    lengths = td[('collector', 'mask')].sum(dim=1)
    truncated_lengths = lengths % segment_length
    num_segments = torch.sum(lengths // segment_length)
    batch_index = torch.repeat_interleave(torch.arange(num_segments), truncated_lengths)
    time_index = torch.arange(segment_length).repeat(num_segments)
    indices = torch.stack([batch_index, time_index], dim=0)

    for k, v in list(tensordict.items()):
        td[k] = torch.zeros(num_segments, segment_length, *v.shape[2:], dtype=v.dtype, device=v.device).scatter_(
            dim=0, index=indices, src=v,
        )

    return td
```

The use case would be something like 
```python
td = split_trajectories(td)
td = truncate_trajectories(td)
```

## Alternatives

We could also add a `max_segment_length` arguments to `split_trajectories` and do this sort of logic within `split_trajectories`.
## Checklist

- [x] I have checked that there is no similar issue in the repo (**required**)

**Comment by vmoens** [comment]
We have a transform for that I think

https://pytorch.org/rl/reference/generated/torchrl.envs.transforms.RandomCropTensorDict.html#torchrl.envs.transforms.RandomCropTensorDict

Is what you want different?

**Comment by smorad** [comment]
If I am understanding this correctly, `RandomCropTensorDict` would just discard anything outside of the window, right? Instead of discarding, I would like the remainder of a sequence to occupy a new row in a `B x T` matrix.
